resolver/addressable_resolver_test.go

```diff
@@ -64,97 +63,99 @@ func TestGetURI_ObjectReference(t *testing.T) {
 		wantURI string
 		wantErr error
 	}{"nil everything": {
-		wantErr: fmt.Errorf("destination missing ObjectReference and URI, expected exactly one"),
-	}, "URI with path": {
+		wantErr: fmt.Errorf("destination missing Ref and URI, expected at least one"),
+	}, "Happy URI with path": {
 		dest: apisv1alpha1.Destination{
 			URI: &apis.URL{
 				Scheme: "http",
 				Host:   "example.com",
 				Path:   "/foo",
 			},
-			Path: ptr.String("/bar"),
 		},
-		wantURI: "http://example.com/foo/bar",
-	}, "URI with path without leading slash": {
-		dest: apisv1alpha1.Destination{
-			URI: &apis.URL{
-				Scheme: "http",
-				Host:   "example.com",
-				Path:   "/foo",
-			},
-			Path: ptr.String("bar"),
-		},
-		wantURI: "http://example.com/foo/bar",
-	}, "URI with garbage path": {
+		wantURI: "http://example.com/foo",
+	}, "URI is not absolute URL": {
 		dest: apisv1alpha1.Destination{
 			URI: &apis.URL{
-				Scheme: "http",
 				Host:   "example.com",
-				Path:   "/foo",
 			},
-			Path: ptr.String("////bar///"),
 		},
-		wantURI: "http://example.com/foo/bar",
-	}, "URI with nil path": {
-		dest: apisv1alpha1.Destination{
-			URI: &apis.URL{
-				Scheme: "http",
-				Host:   "example.com",
-				Path:   "/foo",
+		wantErr: fmt.Errorf("invalid URI: %v", "//example.com"),
+	}, "URI with no host": {
+			dest: apisv1alpha1.Destination{
+				URI: &apis.URL{
+					Scheme: "http",
+				},
 			},
-			Path: nil,
+			wantErr: fmt.Errorf("invalid URI: %v", "http:"),
 		},
-		wantURI: "http://example.com/foo",
-	}, "happy objectref": {
+		"happy ref": {
 		objects: []runtime.Object{
 			getAddressable(),
 		},
-		dest:    apisv1alpha1.Destination{ObjectReference: getAddressableRef()},
+		dest:    apisv1alpha1.Destination{Ref: getAddressableRef()},
 		wantURI: addressableDNS,
-	}, "object ref with path": {
+	}, "ref with relative uri": {
 		objects: []runtime.Object{
 			getAddressable(),
 		},
 		dest: apisv1alpha1.Destination{
-			ObjectReference: getAddressableRef(),
-			Path:            ptr.String("/foo"),
+			Ref: getAddressableRef(),
+			URI: &apis.URL{
+			Path:   "/foo",
+		},
 		},
 		wantURI: addressableDNS + "/foo",
-	}, "object ref with path without leading slash": {
+	}, "ref with relative URI without leading slash": {
```

Comment:
I'd love to see two additional tests that verify the expected behavior for Addressables with paths, both with a trailing slash like `example.com/foo/` and without like `example.com/foo`.